### DISCUSSION TITLE
Training Detectron2 models using data parallel

### DISCUSSION THREAD

**Discussion by AvivSham** [facebookresearch/detectron2#3345] [Q&A]
Hi all,
Is it possible to train Detecrton2 models using data parallel pytroch module (i.e. training model using multiple gpus)?
If not I think this should be high priority feature! since we want to speed up training and use HD images in combination with heavyweight models which won't run on single GPU due to memory issues.

If indeed this is impossible in the current version it there any workaround to make it work?

Thank you in advance.

**Comment by ppwwyyxx** [comment]
The example training script (https://detectron2.readthedocs.io/en/latest/tutorials/getting_started.html#training-evaluation-in-command-line) supports training with >1 GPUs.

**Reply by AvivSham** [reply]
Hi @ppwwyyxx 
Can you please explain in depth what is happening under the hood? I want to speed up my training process by significantly increasing the batch size as I run the training procedure on NVIDIA DGX machine.

**Reply by AvivSham** [reply]
In addition, the link you shared is referring to multi GPU training, what about inference?
